src/test/java/com/wikia/webdriver/testcases/discussions/CreatingReplyTests.java

```diff
@@ -8,73 +8,185 @@
+  public void userOnDesktopShouldFollowPostWhenAddedReplyOnPostDetailsPage() {
+    final PostDetailsPage page = new PostDetailsPage().open(createPostAsUserRemotely().getId());
+    final ReplyCreator replyCreator = page.getReplyCreatorDesktop();
+
+    assertThatPostIsFollowedWhenReplyAdded(page, replyCreator);
+    assertThatPostCanBeUnfollowed(page);
+  }
+
+  // Testing methods
+
+  private void userOnDesktopMustBeLoggedInToUseReplyCreator(final ReplyCreator replyCreator) {
+    anonymousUserOnReplyEditorClickIsRedirectedTo(replyCreator, MercurySubpages.REGISTER_PAGE);
+  }
+
+  private void anonymousUserOnReplyEditorClickIsRedirectedTo(final ReplyCreator replyCreator, final String urlFragment) {
     Assertion.assertTrue(replyCreator.click().isModalDialogVisible());
 
     replyCreator.clickOkButtonInSignInDialog();
-
     Assertion.assertTrue(replyCreator.click().isModalDialogVisible());
 
     replyCreator.clickSignInButtonInSignInDialog();
+    Assertion.assertTrue(driver.getCurrentUrl().contains(urlFragment));
+  }
+
+  private void userOnMobileMustBeLoggedInToUseReplyCreator(final ReplyCreator replyCreator) {
+    anonymousUserOnReplyEditorClickIsRedirectedTo(replyCreator, MercurySubpages.JOIN_PAGE);
+  }
+
+  private PostEntity.Data createPostAsUserRemotely() {
+    return DiscussionsOperations.using(User.USER, driver).cratePostWithUniqueData();
+  }
+
+  private void assertThatUserCanCreateReply(PostDetailsPage page, ReplyCreator replyCreator) {
+    replyCreator.click().clickGuidelinesReadButton();
+    Assertion.assertFalse(replyCreator.isSubmitButtonActive(), SUBMIT_BUTTON_INACTIVE_MESSAGE);
+
+    replyCreator.add(TextGenerator.createUniqueText());
+    Assertion.assertTrue(replyCreator.isSubmitButtonActive(), SUBMIT_BUTTON_ACTIVE_MESSAGE);
+
+    replyCreator.clearText();
+    Assertion.assertFalse(replyCreator.isSubmitButtonActive(), SUBMIT_BUTTON_INACTIVE_MESSAGE);
+
+    final String text = TextGenerator.createUniqueText();
+    replyCreator.add(text).clickSubmitButton();
+    page.getReplies().waitForReplyToAppearWith(text);
+    Assertion.assertFalse(page.getReplies().isEmpty(), REPLY_ADDED_MESSAGE);
+  }
 
-    Assertion.assertTrue(driver.getCurrentUrl().contains(MercurySubpages.JOIN_PAGE));
+  private void assertThatPostIsFollowedWhenReplyAdded(PostDetailsPage page, ReplyCreator replyCreator) {
+    final String text = TextGenerator.createUniqueText();
+    replyCreator.click()
+        .clickGuidelinesReadButton()
+        .add(text)
+        .clickSubmitButton();
+    page.getReplies().waitForReplyToAppearWith(text);
+
+    final PostActionsRow postActions = page.getPost().findNewestPost().findPostActions();
+    Assertion.assertTrue(postActions.isFollowed(), POST_FOLLOWED_BY_DEFAULT);
+  }
+
+  private void assertThatPostCanBeUnfollowed(PostDetailsPage page) {
+    final PostActionsRow postActions = page.getPost().findNewestPost().findPostActions();
+    postActions.clickFollow();
+    sleepForOneSecond();
```

Comment:
Can we move `sleepForOneSecond` method inside `clickFollow`? As I understood from our previous discussion, explicit wait is always necessary in case of following a post. My request also includes following tests